src/rgw/rgw_op.cc

```diff
@@ -2286,12 +2286,12 @@ void RGWGetObj::execute(optional_yield y)
   }
 
   op_ret = read_op->prepare(s->yield, this);
-  if (op_ret < 0)
-    goto done_err;
   version_id = s->object->get_instance();
   s->obj_size = s->object->get_obj_size();
   attrs = s->object->get_attrs();
   multipart_parts_count = read_op->params.parts_count;
+  if (op_ret < 0)
```

Comment:
what about other error codes? is it safe to get attributes, size etc, if the head  read failed for some other reason?
maybe wrap with `if (op_ret == 0 || op_ret == -ERR_NOT_MODIFIED)` ?

Reply:
the variables are safe to copy on a read error, they'd just have their default values